llmsherpa/readers/layout_reader.py

```diff
@@ -456,130 +613,107 @@ def read(self, blocks_json):
                 parent = node
             prev_node = node
 
         return root
 
+
 class Document:
     """
     A document is a tree of blocks. It is the root node of the layout tree.
     """
+
     def __init__(self, blocks_json):
         self.reader = LayoutReader()
         self.root_node = self.reader.read(blocks_json)
         self.json = blocks_json
-        self.top_sections = self._get_top_sections()
+
     def chunks(self):
         """
         Returns all the chunks in the document. Chunking automatically splits the document into paragraphs, lists, and tables without any prior knowledge of the document structure.
         """
         return self.root_node.chunks()
+
     def tables(self):
         """
         Returns all the tables in the document. This is useful for getting all the tables in a document.
         """
         return self.root_node.tables()
+
     def sections(self):
         """
         Returns all the sections in the document. This is useful for getting all the sections in a document.
         """
         return self.root_node.sections()
-    
-    def to_text(self, include_duplicates = False):
+
+    def to_text(self):
         """
-        Returns text of a document by iterating through all the sections '\n'
-        :param include_duplicates: bool
-            If True, then text of all the sections is included. If False, then only the text of the top sections is included.
+        Returns text of a document by iterating through all the children
         """
         text = ""
-
-        if include_duplicates:
-            for section in self.sections():
-                text = text + section.to_text(include_children=True, recurse=True) + "\n"
-        else:
-            for section in self.top_sections:
-                text = text + section.to_text(include_children=True, recurse=True) + "\n"
-
+        for child in self.root_node.children:
+            text += child.to_text(include_children=True, recurse=True) + "\n"
         return text
-                   
-    def to_html(self, include_duplicates = False):
+
+    def to_html(self):
         """
-        Returns html for the document by iterating through all the sections
-        :param include_duplicates: bool
-            If True, then html of all the sections is included. If False, then only the html of the top sections is included.
+        Returns html for the document by iterating through all the children
         """
         html_str = "<html>"
-
-        if include_duplicates:
-            for section in self.sections():
-                html_str = html_str + section.to_html(include_children=True, recurse=True)
-        else:
-            for section in self.top_sections:
-                html_str = html_str + section.to_html(include_children=True, recurse=True)
-
-        html_str = html_str + "</html>"
+        for child in self.root_node.children:
+            html_str += child.to_html(include_children=True, recurse=True)
```

Comment:
Thank you for this fix, I was trying to figure out why I was seeing duplicate sections in my HTML.

Reply:
you're welcome!